cmd/activity-monitor/main.go

```diff
@@ -88,13 +90,43 @@ func startMonitor(rpcCh *amqp.Channel, logger *blog.AuditLogger) {
 		cmd.FailOnError(err, "Could not subscribe to queue")
 	}
 
+	deliveryTimings := make(map[string]time.Time)
+
 	// Run forever.
 	for d := range deliveries {
+		// If d is a call add to deliveryTimings and increment Boulder.RpcCallsWaiting, if it is a 
+		// response then get time.Since original call from deliveryTiming, send timing metric, and
+		// decrement Boulder.RpcCallsWaiting
+		go func() {
+			if d.ReplyTo != "" {
+				deliveryTimings[fmt.Sprintf("%s:%s", d.CorrelationId, d.ReplyTo)] = time.Now()
+				if err := stats.Inc("RpcCallsOpen", 1, 1.0); err != nil {
+					logger.Warning(fmt.Sprintf("Could not send stats to Statsd server: %s", err))
+				}
+			} else {
+				rpcSent := deliveryTimings[fmt.Sprintf("%s:%s", d.CorrelationId, d.RoutingKey)]
+				if rpcSent != *new(time.Time) {
+					respTime := time.Since(rpcSent)
+					delete(deliveryTimings, fmt.Sprintf("%s:%s", d.CorrelationId, d.RoutingKey))
+
+					if err := stats.TimingDuration(fmt.Sprintf("RpcCallTime.%s", d.Type), respTime, 1.0); err != nil {
+						logger.Warning(fmt.Sprintf("Could not send stats to Statsd server: %s", err))
+					}
+					if err := stats.Dec("RpcCallsOpen", 1, 1.0); err != nil {
```

Comment:
Similar advice to the below; Inc/Dec never seems to stay put at 0 in production on statsd.